### ISSUE TITLE
Duplicated download exceptions in 6.5

### ISSUE THREAD

**Issue by colin1497** [Tribler/tribler#2018]
This is a log file from the 6.5 release for review.
- [N] I have tried with the latest pre-release version and I still can reproduce the issue.
##### Relevant log file output:

[tribler.exe.log.txt](https://github.com/Tribler/tribler/files/165746/tribler.exe.log.txt)

EDIT(whirm): Relevant portion of the log:

```
Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run

  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner

  File "twisted\internet\base.pyo", line 1192, in run

  File "twisted\internet\base.pyo", line 1201, in mainLoop

--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 824, in runUntilCurrent

  File "Tribler\Core\Libtorrent\LibtorrentDownloadImpl.pyo", line 201, in <lambda>

  File "Tribler\Core\Libtorrent\LibtorrentDownloadImpl.pyo", line 248, in network_create_engine_wrapper

  File "Tribler\Core\Libtorrent\LibtorrentMgr.pyo", line 254, in add_torrent

Tribler.Core.exceptions.DuplicateDownloadException: <class 'Tribler.Core.exceptions.DuplicateDownloadException'>: None
ERROR   1457364316.25             log:443   Unhandled Error
Traceback (most recent call last):
  File "threading.pyo", line 763, in run

  File "Tribler\Core\Utilities\twisted_thread.pyo", line 55, in _reactor_runner

  File "twisted\internet\base.pyo", line 1192, in run

  File "twisted\internet\base.pyo", line 1201, in mainLoop

--- <exception caught here> ---
  File "twisted\internet\base.pyo", line 824, in runUntilCurrent

  File "Tribler\Core\Libtorrent\LibtorrentDownloadImpl.pyo", line 201, in <lambda>

  File "Tribler\Core\Libtorrent\LibtorrentDownloadImpl.pyo", line 248, in network_create_engine_wrapper

  File "Tribler\Core\Libtorrent\LibtorrentMgr.pyo", line 251, in add_torrent

exceptions.RuntimeError: torrent already exists in session

```

**Comment by whirm** [comment]
Thanks for the report!

**Comment by colin1497** [comment]
No prob, trying to use the release version for a while before I step back into dev builds.

**Comment by whirm** [comment]
@colin1497 Can you reproduce it at will? I can't seem to reproduce it on current `next`

**Comment by colin1497** [comment]
I have not tried to reproduce.  I'm not even actually certain what the original cause was.  Really I just got a log file at shut down, knew that something somewhere had failed and thought you might find it interesting.  Most likely it's just some condition you need to trap and ignore, but I'm not sure what I did.  Maybe clicked a magnet link twice without completing the dialog?

**Comment by whirm** [comment]
I don't really want to ignore it, I would like to fix the root cause of it. 

I've tried to reproduce it again for a bit and also tried the magnet link thing and it didn't break. So for now I'll push it to the V6.6 milestone and see if we can get anybody to figure out how to reproduce this.

Cheers!

**Comment by devos50** [comment]
I also can't reproduce it. I will push it to the next release.

**Comment by devos50** [comment]
I suggest to try the following: start a download and play it using the video player. Make sure that you start the download with `hidden seeding` enabled (anon download is optional but for faster reproduction, I would disable this). After the download of the torrent is finished, the torrent gets removed and re-added to the libtorrent engine, making our video server confused.

**Comment by qstokkink** [comment]
This was fixed in #3062.
